gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/VirtualHostServiceImpl.java

```diff
@@ -71,8 +72,28 @@ public Collection<VirtualHost> sanitizeAndValidate(Collection<VirtualHost> virtu
         // Sanitize virtual hosts
         Collection<VirtualHost> sanitizedVirtualHosts = virtualHosts.stream().map(this::sanitize).collect(Collectors.toList());
 
+        final EnvironmentEntity currentEnv = environmentService.findById(GraviteeContext.getCurrentEnvironment());
+
         // validate domain restrictions
-        validateDomainRestrictions(sanitizedVirtualHosts);
+        validateDomainRestrictions(sanitizedVirtualHosts, currentEnv.getDomainRestrictions());
+
+        // In virtual host mode, every vhost should have a host set
+        final boolean virtualHostModeEnabled =
+            sanitizedVirtualHosts.size() > 1 ||
+            sanitizedVirtualHosts.iterator().next().getHost() != null ||
+            !currentEnv.getDomainRestrictions().isEmpty();
```

Comment:
I think we need an additional nullcheck here.
Tested it locally, it throws an NPE cause `currentEnv.getDomainRestrictions()` is null.

Reply:
done :), good catch